### PR TITLE
using theme color

### PR DESCRIPTION
**Pull request by emre2038**
## Description
coloring 
* the theme icon,
* selected theme & category background 
* point markers
* clusters

according to the selected color in the theme configuration page in the admin panel

## Related Issue
https://wazimap.atlassian.net/browse/WNCM-299

## How is it tested automatically?
added automated tests into `__tests__/gui/point_data.feature` and `__tests__/utils.test.js`

## How should a reviewer test it locally
* go to admin panel
* you can use the review app https://wazimap-pr-424.herokuapp.com/admin
* in the theme configuration page change the color
* refresh the map and confirm that the theme color is applied

## Screenshots
image

## Checklist

- [x]  🚀 is the code ready to be merged and go live?
- [x]  🛠 does it work (build) locally
- [x] 👩‍🎨 does the design match the [Demo](https://wazimap-ng-v1.webflow.io/demo)

### Pull Request

- [x]  📰 good title
- [x]  📝good description
- [x]  🔖 issue linked
- [x]  📖 changelog filled out
- [x] commit messages are meaningful

### Code Quality

- [x]  🚧 no commented out code
- [x]  🖨 no unnecessary logging
- [x]  🎱 no magic numbers

### Testing

- [x]  ✅ added (appropriate) unit tests
- [x]  💢 edge cases in tests were considered
- [x]  ✅ ran tests locally & are passing

### CODE REVIEW

File: src/js/elements/point_tray/theme.js [hunk 1]
```diff
@@ -103,4 +108,13 @@ export class Theme extends Component {
     get id() {
         return this.data.id;
     }
+
+    get color() {
+        return this.data.color;
+    }
+
+    get backgroundColor() {
+        console.log({'theme': this.data})
```
Comment (michaelglenister) [inline src/js/elements/point_tray/theme.js:117]: Can this log be removed?

File: src/js/elements/point_tray/theme.js [hunk 2]
```diff
@@ -29,12 +28,15 @@ export class Theme extends Component {
         $(this.element).find('.point-mapper__h1_checkbox input[type=checkbox]').on('click', () => self.toggle());
         $('.point-data__h1_name .truncate', this.element).text(this.name);
         $('.point-mapper__h1_trigger', this.element)
-            .removeClass(activeClsName)
-            .addClass('theme-' + this.themeIndex)
+            .removeClass('active')
+            .addClass('theme-' + this.themeIndex)   //remove this line
```
Comment (michaelglenister) [inline src/js/elements/point_tray/theme.js:32]: Was this comment a todo?

File: src/js/elements/point_tray/category.js [hunk 3]
```diff
@@ -135,4 +140,13 @@ export class Category extends Component {
     get theme() {
         return this.data.theme;
     }
+
+    get color() {
+        return this.data.theme.color;
+    }
+
+    get backgroundColor() {
+        console.log({'category': this.data})
```
Comment (michaelglenister) [inline src/js/elements/point_tray/category.js:149]: Is this log needed?

File: __tests__/utils.test.js [hunk 4]
```diff
@@ -1,120 +1,126 @@
...
+
+        expect(filledObject.first).toBe(1)
+        expect(filledObject.second.second_1).toBe(4)
+        expect(filledObject.second.second_2.second_2_1).toBe(5)
+    })
+
+    test('check that mid-color is calculated correctly', () => {
+        let midColor = calculateMidColor('#ffffff', '#22a6b3');
+
+        expect(midColor).toBe('#90D2D9');
```
Comment (michaelglenister) [inline __tests__/utils.test.js:123]: Unit test for calculateMidColor, looks good

**Review by michaelglenister — COMMENTED** [review]
A few minor comments otherwise it looks good

File: src/js/elements/point_tray/theme.js (hunk 1 above)
Comment (emre2038) [reply src/js/elements/point_tray/theme.js]: removed 👍

File: src/js/elements/point_tray/category.js (hunk 3 above)
Comment (emre2038) [reply src/js/elements/point_tray/category.js]: removed 👍

File: __tests__/utils.test.js (hunk 4 above)
Comment (emre2038) [reply __tests__/utils.test.js:123]: thanks 👍

File: src/js/elements/point_tray/theme.js (hunk 2 above)
Comment (emre2038) [reply src/js/elements/point_tray/theme.js]: I knew I was forgetting something :) yes it was a todo and removed it. nice catch

**Pull request merged**